server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/VariableWidthHistogramAggregationBuilder.java

```diff
@@ -149,12 +158,32 @@ protected ValuesSourceAggregatorFactory innerBuild(QueryShardContext queryShardC
                                                        ValuesSourceConfig config,
                                                        AggregatorFactory parent,
                                                        AggregatorFactories.Builder subFactoriesBuilder) throws IOException {
-
         Settings settings = queryShardContext.getIndexSettings().getNodeSettings();
         int maxBuckets = MultiBucketConsumerService.MAX_BUCKET_SETTING.get(settings);
         if (numBuckets > maxBuckets) {
-            throw new IllegalArgumentException(NUM_BUCKETS_FIELD.getPreferredName()+
-                " must be less than " + maxBuckets);
+            throw new IllegalArgumentException(NUM_BUCKETS_FIELD.getPreferredName() + " must be less than " + maxBuckets);
+        }
+        int initialBuffer = getInitialBuffer();
+        int shardSize = getShardSize();
+        if (initialBuffer < numBuckets) {
+            // If numBuckets buckets are being returned, then at least that many must be stored in memory
+            throw new IllegalArgumentException(
```

Comment:
This is pretty late in the process... is there a reason we can't check these when parsing the request?  I don't see anything that needs the shard context but I might be missing something?

Reply:
Yeah, it is. We don't really have a validate stage for these and I didn't really want to do it in the parsing because I figured it was possible for it to "escape" validation sometimes.